pkg/routeimporter/http.go

```diff
@@ -0,0 +1,73 @@
+package routeimporter
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/blakewilliams/viewproxy"
+)
+
+func LoadHttp(ctx context.Context, server *viewproxy.Server, path string) error {
+	var routeEntries []ConfigRouteEntry
+
+	target, err := url.Parse(server.Target())
+
+	if err != nil {
+		return fmt.Errorf("could not parse target: %w", err)
+	}
+
+	target.Path = path
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, target.String(), nil)
+
+	if err != nil {
+		return fmt.Errorf("Could not create a request when loading config: %w", err)
+	}
+
+	if server.HmacSecret != "" {
+		setHmacHeaders(req, server.HmacSecret)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+
+	if err != nil {
+		return fmt.Errorf("could not fetch JSON configuration: %w", err)
+	}
+
+	routesJson, err := io.ReadAll(resp.Body)
+
+	if err != nil {
+		return fmt.Errorf("could not read route config response body: %w", err)
+	}
+
+	if err := json.Unmarshal(routesJson, &routeEntries); err != nil {
+		return fmt.Errorf("could not unmarshal route config json: %w", err)
+	}
+
+	err = LoadRoutes(server, routeEntries)
+
+	if err != nil {
```

Comment:
This looks like it could be a nice one liner like the `json.Unmarshal` line:

```suggestion
	if err = LoadRoutes(server, routeEntries); err != nil {
```

Reply:
Nice!